iml-services/iml-snapshot/src/main.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use futures_util::stream::TryStreamExt;
+use iml_manager_env::get_pool_limit;
+use iml_postgres::{get_db_pool, sqlx};
+use iml_service_queue::service_queue::consume_data;
+use iml_tracing::tracing;
+use iml_wire_types::snapshot;
+
+// Default pool limit if not overridden by POOL_LIMIT
+const DEFAULT_POOL_LIMIT: u32 = 2;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    iml_tracing::init();
+
+    let pool = iml_rabbit::connect_to_rabbit(1);
+
+    let conn = iml_rabbit::get_conn(pool).await?;
+
+    let ch = iml_rabbit::create_channel(&conn).await?;
+
+    let mut s = consume_data::<Vec<snapshot::Snapshot>>(&ch, "rust_agent_snapshot_rx");
+
+    let pool = get_db_pool(get_pool_limit().unwrap_or(DEFAULT_POOL_LIMIT)).await?;
+    sqlx::migrate!("../../migrations").run(&pool).await?;
+
+    while let Some((fqdn, snapshots)) = s.try_next().await? {
+        tracing::debug!("snapshots from {}: {:?}", fqdn, snapshots);
+
+        let snaps = snapshots.into_iter().fold(
+            (vec![], vec![], vec![], vec![], vec![], vec![], vec![]),
+            |mut acc, s| {
+                let d = &s.details[0];
+
+                acc.0.push(s.filesystem_name);
+                acc.1.push(s.snapshot_name);
+                acc.2.push(d.create_time.naive_utc());
+                acc.3.push(d.modify_time.naive_utc());
+                acc.4.push(d.snapshot_fsname.clone());
+                acc.5.push(d.status == Some(snapshot::Status::Mounted));
+                acc.6.push(d.comment.clone());
```

Comment:
We should save `snapshot_role` as well, otherwise it's not possible to read `Detail` back.

Reply:
`snapshot_role` is not reported by `lctl` by default, though I believe it would MDT.

You can compare [default](https://github.com/whamcloud/integrated-manager-for-lustre/blob/952361e231351ab8e22661c7d89bd49b38dcecad/iml-agent/src/action_plugins/lustre/snapshot/parse.rs#L193) and [detailed](https://github.com/whamcloud/integrated-manager-for-lustre/blob/952361e231351ab8e22661c7d89bd49b38dcecad/iml-agent/src/action_plugins/lustre/snapshot/parse.rs#L236) output.

Reply:
Can't you just do this?

```
        let snaps = snapshots.into_iter().fold(
            (vec![], vec![], vec![], vec![], vec![], vec![], vec![], vec![]),
            |mut acc, s| {
                let d = &s.details[0];

                acc.0.push(s.filesystem_name);
                acc.1.push(s.snapshot_name);
                acc.2.push(d.create_time.naive_utc());
                acc.3.push(d.modify_time.naive_utc());
                acc.4.push(d.snapshot_fsname.clone());
                acc.5.push(d.status == Some(snapshot::Status::Mounted));
                acc.6.push(d.comment.clone());
                acc.7.push(d.snapshot_role.clone());
```

Reply:
There is no need. If we decide to support detailed snapshots, we will need different DB schema, like two tables.

Reply:
But you're already writing some of the details to the same table. Why not just write all the fields there? I don't understand what's the difference.

Reply:
Maybe we remove the field from the struct for now, then?

Reply:
There isn't enough data to decide yet, but I'd be more inclined to changing the Snapshot structure rather than the DB scheme.

All components of a snapshot are design to have the same timestamps, comments, etc, although technically they all can be different. So there is little sense to keep snapshot_role around at all, IMHO.